File: android/src/main/kotlin/com/fidbek/flutter/FidbekFlutterPlugin.kt
package com.fidbek.flutter

import android.app.Application
import com.fidbek.sdk.Fidbek
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

class FidbekFlutterPlugin : FlutterPlugin, MethodCallHandler {
    private lateinit var channel: MethodChannel
    private lateinit var application: Application

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        application = binding.applicationContext as Application
        channel = MethodChannel(binding.binaryMessenger, "fidbek_flutter")
        channel.setMethodCallHandler(this)
    }

    override fun onMethodCall(call: MethodCall, result: Result) {
        when (call.method) {
            "configure" -> {
                val token = call.argument<String>("token")
                if (token.isNullOrBlank()) {
                    result.error("INVALID_ARGUMENT", "token is required", null)
                    return
                }

                val shakeToOpenEnabled = call.argument<Boolean>("shakeToOpenEnabled") ?: true
                Fidbek.initialize(application, token, shakeToOpenEnabled)
                result.success(null)
            }

            "open" -> {
                Fidbek.open()
                result.success(null)
            }

            "identify" -> {
                val userId = call.argument<String>("userId")
                val name = call.argument<String>("name")
                val email = call.argument<String>("email")
                if (userId.isNullOrBlank()) {
                    if (name.isNullOrBlank() && email.isNullOrBlank()) {
                        result.error(
                            "INVALID_ARGUMENT",
                            "At least one of userId, name, or email is required",
                            null
                        )
                        return
                    }
                }

                Fidbek.identify(
                    userId = userId,
                    name = name,
                    email = email
                )
                result.success(null)
            }

            "clearIdentity" -> {
                Fidbek.clearIdentity()
                result.success(null)
            }

            "shutdown" -> {
                Fidbek.shutdown()
                result.success(null)
            }

            else -> result.notImplemented()
        }
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }
}

File: ios/Classes/FidbekFlutterPlugin.swift
import FidbekSDK
import Flutter
import UIKit

@objc(FidbekFlutterPlugin)
public class FidbekFlutterPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "fidbek_flutter", binaryMessenger: registrar.messenger())
    let instance = FidbekFlutterPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "configure":
      guard let args = call.arguments as? [String: Any],
            let token = args["token"] as? String,
            !token.isEmpty else {
        result(
          FlutterError(
            code: "INVALID_ARGUMENT",
            message: "token is required",
            details: nil
          )
        )
        return
      }

      let shakeToOpenEnabled = args["shakeToOpenEnabled"] as? Bool ?? true
      Fidbek.shared.configure(token: token, shakeToOpenEnabled: shakeToOpenEnabled)
      result(nil)

    case "open":
      Fidbek.shared.present()
      result(nil)

    case "identify":
      guard let args = call.arguments as? [String: Any] else {
        result(
          FlutterError(
            code: "INVALID_ARGUMENT",
            message: "identify arguments are required",
            details: nil
          )
        )
        return
      }

      let userId = (args["userId"] as? String)?.trimmingCharacters(in: .whitespacesAndNewlines)
      let name = (args["name"] as? String)?.trimmingCharacters(in: .whitespacesAndNewlines)
      let email = (args["email"] as? String)?.trimmingCharacters(in: .whitespacesAndNewlines)
      if (userId?.isEmpty ?? true) && (name?.isEmpty ?? true) && (email?.isEmpty ?? true) {
        result(
          FlutterError(
            code: "INVALID_ARGUMENT",
            message: "At least one of userId, name, or email is required",
            details: nil
          )
        )
        return
      }

      Fidbek.shared.identify(
        userId: userId,
        name: name,
        email: email
      )
      result(nil)

    case "clearIdentity":
      Fidbek.shared.clearIdentity()
      result(nil)

    case "shutdown":
      Fidbek.shared.stop()
      result(nil)

    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
